### DISCUSSION TITLE
Parsing Swagger UI

### DISCUSSION THREAD

**Discussion by 0xnur** [projectdiscovery/katana#455] [Ideas]
Should katana support and parse specific UI's such as SWAGGER? 

It can be used with a known file parameters or a new parameters such as known page.

https://petstore.swagger.io

**Comment by MetzinAround** [comment]
I'm not sure how Katana works under the hood, but I'll tag @ehsandeep to ask if this is possible/something that has been thought about.

**Comment by ehsandeep** [comment]
@0xnur For sure it would be useful for katana to support parsing swagger API specs, currently it does not!

we have plans to support fuzzing in [nuclei](https://github.com/projectdiscovery/nuclei) which will require supporting input / parsing multiple well-known API specs such as

- OpenAPI
- GraphQL
- Swagger
- PostMan
- SOAP

so it's something we can work and reuse as common code across multiple pd projects including katana / nuclei.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
